spirv_msl.cpp

```diff
@@ -2491,18 +2484,43 @@ string CompilerMSL::image_type_glsl(const SPIRType &type)
 	}
 
 	// Append the pixel type
-	auto &img_pix_type = get<SPIRType>(img_type.type);
 	img_type_name += "<";
-	img_type_name += type_to_glsl(img_pix_type);
+	img_type_name += type_to_glsl(get<SPIRType>(img_type.type));
 
-	if (img_type.is_written)
+	// For unsampled images, append the sample/read/write access qualifier.
+	// For kernel images, the access qualifier my be supplied directly by SPIR-V.
+	// Otherwise it may be set based on whether the image is read from or written to within the shader.
+	if (type.basetype == SPIRType::Image)
```

Comment:
What it means here is that I'm not sure I want to ignore `OpTypeImage` _Sampled_ == 0.